pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -534,10 +534,23 @@ func translateHeaderMatch(name string, in *networking.StringMatch) route.HeaderM
 func translateRetryPolicy(in *networking.HTTPRetry) *route.RouteAction_RetryPolicy {
 	if in != nil && in.Attempts > 0 {
 		d := util.GogoDurationToDuration(in.PerTryTimeout)
+		// default retry on condition
+		retryOn := "gateway-error"
```

Comment:
Do you want to force default retries for `gateway-error`?

Reply:
That's @rshriram suggested https://github.com/istio/istio/issues/8081#issuecomment-419092492

Reply:
Sorry. actually I meant asking are you defaulting only for `gateway-error`? I think some of the gRPC codes like for example `unavailable ` may be considered for default so that every user/deployment need not have to specify it. NBD though.

Reply:
That’s a good point. What do you think are the common conditions for which we should have a default retry policy?
Also by default I mean the user specifying a retry setting in the virtual service with no retry policy.

Reply:
I think this should be connect failure, refused stream, gateway error, unavailable.

Reply:
How about including cancelled and resource-exhausted by default ?

Reply:
sure. Keep in mind that the retries will kick in if and only if you specify the retry setting in the virtual service. At that point you have the option of going with the default retry policies or specifying your own.